tree/master/cloud/src/deployments/Static Content/Matter Center Assets/Document Landing Assets/Scripts/documentDetails.js

```diff
@@ -599,6 +600,83 @@ var oDocumentLandingObject = (function () {
             }
         }
     };
+    
+    oCommonFunctions.retrieveAdditionalMatterPropertyContentType = function (oData) {
+        // Wait for getCurrentUserTitle to complete
+        if (!oGlobalConstants.sUserLoginName) {
+            setTimeout(function () { oCommonFunctions.retrieveListItems(); }, 1000);
+        } else {
+            if (oGlobalConstants.sUserLoginName) {
+   				getallContentTypes(oData);
+   	        } else {
+                onQueryFailed(oGlobalConstants.sOperationPin);
+            }
+        }
+    };
+    
+    function getallContentTypes(oData) {
+		var clientContext = SP.ClientContext.get_current();
+		var oContentTypes = clientContext.get_web().get_availableContentTypes();
+		clientContext.load(oContentTypes);
+		clientContext.executeQueryAsync(
+			Function.createDelegate(this, function() {
+				var ctypesInfo = '';
+				var ctypesEnumerator = oContentTypes.getEnumerator();
+			
+				while (ctypesEnumerator.moveNext()) {
+					var ocontentType = ctypesEnumerator.get_current();
+					if(ocontentType.get_name() == uiconfigs.DocumentDetails.AdditionalContentTypeName)
+					{
+						getAllFieldsInContentType(ocontentType,oData);
+					}
+				}
+				console.log(ctypesInfo.toString());
+			}),
+			Function.createDelegate(this, function() {
+				console.log('failed');
+			})
+		);
+	};
+
+function getAllFieldsInContentType(contentTypeName,oData){
+    var isDispalyExtraProperties = false;
```

Comment:
Purpose: To retrieve all site columns Internal Name and Display Name of specific content type and generating HTML to display it.